src/app/core/error-page/error-page.component.spec.ts

```diff
@@ -0,0 +1,82 @@
+import {ErrorPageComponent} from './error-page.component';
+import {async, TestBed} from '@angular/core/testing';
+import {ComponentFixture} from '@angular/core/testing/src/component_fixture';
+import {ActivatedRoute} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {Observable} from 'rxjs/Observable';
+
+class MockActivatedRoute {
+  data: Observable<{ errorCode: string }>;
+
+  constructor(errorCode: string) {
+    this.data = Observable.of({errorCode});
+  }
+}
+
+
+function setup(activatedRouteMock): {
```

Comment:
Its a different school...the main benefit is that concurrency is (theoretically) possible on a method level. Also, all initialization is explicit instead of "magically" populated via `beforeEach`.
It's mostly a matter of taste...